csrc/python_frontend/python_bindings.cpp

```diff
@@ -3266,5 +3308,63 @@ void initNvFuserPythonBindings(PyObject* module) {
             !isResharding(tv->definition()));
       },
       py::arg("tensor"));
+  nvf_sched.def(
+      "can_schedule",
+      [](FusionDefinition::SchedOperators& self,
+         const ScheduleHeuristic& heuristic) {
+        NVF_CHECK(
+            self.validUse(),
+            "Attempting to use a SchedOperators Op prior to definition!");
+        return canSchedule(self.fusion_definition->userSchedule(), heuristic);
+      },
+      py::arg("heuristic"));
+  nvf_sched.def(
+      "find_compatible_schedulers", [](FusionDefinition::SchedOperators& self) {
+        NVF_CHECK(
+            self.validUse(),
+            "Attempting to use a SchedOperators Op prior to definition!");
+
+        Fusion* fusion = self.fusion_definition->userSchedule()->schedule.get();
+        NVF_ERROR(
+            fusion != nullptr, "Requires Fusion to use heuristic schedulers");
+
+        SchedulerRuntimeInfo* runtime_info =
+            self.fusion_definition->userSchedule()->runtime_info.get();
+        NVF_ERROR(
+            runtime_info != nullptr,
+            "Requires SchedulerRuntimeInfo to use heuristic schedulers");
+
+        std::vector<ScheduleHeuristic> valid_heuristics;
+        valid_heuristics.reserve(all_heuristics_in_priority_order.size());
+        std::copy_if(
+            all_heuristics_in_priority_order.begin(),
+            all_heuristics_in_priority_order.end(),
+            std::back_inserter(valid_heuristics),
+            [fusion, runtime_info](ScheduleHeuristic heuristic) {
+              return SchedulerEntry::canSchedule(
+                  heuristic, fusion, *runtime_info);
+            });
+        return valid_heuristics;
+      });
+  nvf_sched.def(
+      "schedule",
+      [](FusionDefinition::SchedOperators& self,
+         const ScheduleHeuristic& heuristic) {
+        NVF_CHECK(
+            self.validUse(),
+            "Attempting to use a SchedOperators Op prior to definition!");
+        UserSchedule* sched = self.fusion_definition->userSchedule();
+        NVF_CHECK(
+            sched->heuristic_scheduler == nullptr,
+            "Heuristic Scheduler is already defined for this UserSchedule");
```

Comment:
nitpick: would be nice to have the stages of a UserSchedule be exposed as a method, rather than query on each field being initialized.

i.e. `canBeScheduled` (when we check for fusion/runtime info),  `hasBeenScheduled` (when we check for heuristic_schedule).

Reply:
I created `UserSchedule::scheduleWithHeuristic` to build the heuristic and schedule the fusion.